core/arch/arm/plat-imx/drivers/tzc380.c

```diff
@@ -2,36 +2,104 @@
 /*
  * Copyright 2019 Pengutronix
  * All rights reserved.
+ * Copyright 2019 NXP
  *
  * Rouven Czerwinski <entwicklung@pengutronix.de>
  */
 
 #include <drivers/tzc380.h>
 #include <imx-regs.h>
 #include <imx.h>
+#include <kernel/panic.h>
 #include <mm/core_memprot.h>
 #include <mm/generic_ram_layout.h>
 
+#if (defined(PLATFORM_FLAVOR_mx6solosabresd) \
+	|| defined(PLATFORM_FLAVOR_mx6solosabreauto) \
+	|| defined(PLATFORM_FLAVOR_mx6sxsabresd) \
+	|| defined(PLATFORM_FLAVOR_mx6sxsabreauto) \
+	|| defined(PLATFORM_FLAVOR_mx6slevk) \
+	|| defined(PLATFORM_FLAVOR_mx6sllevk) \
+	|| defined(PLATFORM_FLAVOR_mx6ulevk) \
+	|| defined(PLATFORM_FLAVOR_mx6ul9x9evk) \
+	|| defined(PLATFORM_FLAVOR_mx6ullevk) \
+	|| defined(PLATFORM_FLAVOR_mx7dsabresd))
 void imx_configure_tzasc(void)
 {
+	uint8_t region = 1;
+	vaddr_t addr = core_mmu_get_va(TZASC_BASE, MEM_AREA_IO_SEC);
 
-	int i;
-	vaddr_t addr[2] = {0};
+	if (!addr)
+		panic("Cannot get TZASC base address");
+
+	tzc_init(addr);
+
+	region = tzc_auto_configure(CFG_DRAM_BASE, CFG_DDR_SIZE,
+					TZC_ATTR_SP_NS_RW, region);
+	region = tzc_auto_configure(CFG_TZDRAM_START, CFG_TZDRAM_SIZE,
+					TZC_ATTR_SP_S_RW, region);
+	region = tzc_auto_configure(CFG_SHMEM_START, CFG_SHMEM_SIZE,
+					TZC_ATTR_SP_ALL, region);
+
+	DMSG("Action register: %xl", tzc_get_action());
```

Comment:
s/`%xl`/`%" PRIx32 "l"`/   (whatt's `l` for?)

Reply:
It comes from the previous implementation of imx_configure_tzasc()
I don't know why it's there.
I replaced it with `0x%" PRIx32`

Addressed in 03c1e4790198c7f0f064b28447ef8ce0e637376e

Reply:
> s/`%xl`/`%" PRIx32 "l"`/ (whatt's `l` for?)
for `long` which is correct for ARM32 bit platforms, but not for 64-bit. `PRIx32` is the correct choice.

Reply:
Isn't it `%lx` for long?

Reply:
Indeed. This will print `2000l`.